assets/js/modules/reader-revenue-manager/components/PublicationApprovedOverlayNotification.test.js

```diff
@@ -0,0 +1,135 @@
+	RRM_PUBLICATION_APPROVED_OVERLAY_NOTIFICATION,
+} from './PublicationApprovedOverlayNotification';
+import { CORE_UI } from '../../../googlesitekit/datastore/ui/constants';
+import {
+	VIEW_CONTEXT_MAIN_DASHBOARD,
+	VIEW_CONTEXT_MAIN_DASHBOARD_VIEW_ONLY,
+} from '../../../googlesitekit/constants';
+import { Provider as ViewContextProvider } from '../../../components/Root/ViewContextContext';
+import { UI_KEY_READER_REVENUE_MANAGER_SHOW_PUBLICATION_APPROVED_NOTIFICATION } from '../datastore/constants';
+import { CORE_USER } from '../../../googlesitekit/datastore/user/constants';
+
+describe( 'PublicationApprovedOverlayNotification', () => {
+	let registry;
+
+	beforeAll( () => {
+		registry = createTestRegistry();
+	} );
+
+	beforeEach( () => {
+		registry
+			.dispatch( CORE_UI )
+			.setValue(
+				UI_KEY_READER_REVENUE_MANAGER_SHOW_PUBLICATION_APPROVED_NOTIFICATION,
+				true
+			);
+	} );
+
+	it( 'should render the component with correct title and description', async () => {
+		// No items are dismissed.
+		registry.dispatch( CORE_USER ).receiveGetDismissedItems( [] );
+
+		const { getByText, waitForRegistry } = render(
+			<ViewContextProvider value={ VIEW_CONTEXT_MAIN_DASHBOARD }>
+				<PublicationApprovedOverlayNotification />
+			</ViewContextProvider>,
+			{
+				registry,
+			}
+		);
+
+		await waitForRegistry();
+
+		// Make sure that title is rendered.
+		expect(
+			getByText( /Your Reader Revenue Manager publication is approved/ )
+		).toBeInTheDocument();
+
+		// Make sure that description is present in the component.
+		expect(
+			getByText(
+				/Unlock your full reader opportunity by enabling features like subscriptions, contributions and newsletter sign ups/
+			)
+		).toBeInTheDocument();
+
+		// Make sure that `googlesitekit--overlay-notification--reader-revenue-manager` class is present for the component wrapper.
+		expect(
+			document.querySelector(
+				'.googlesitekit--overlay-notification--reader-revenue-manager'
+			)
+		).toBeInTheDocument();
+	} );
+
+	it( 'should return null when dashboard is not main dashboard', async () => {
+		registry.dispatch( CORE_USER ).receiveGetDismissedItems( [] );
+
+		const { container, waitForRegistry } = render(
+			<ViewContextProvider
+				value={ VIEW_CONTEXT_MAIN_DASHBOARD_VIEW_ONLY }
+			>
+				<PublicationApprovedOverlayNotification />
+			</ViewContextProvider>,
+			{
+				registry,
+			}
+		);
+
+		await waitForRegistry();
+
+		// Container should be empty.
+		expect( container ).toBeEmptyDOMElement();
+		// Component should return null.
+		expect( container.firstChild ).toBeNull();
+	} );
+
+	it( 'should return null when notification is dismissed', async () => {
+		// Set the notification as dismissed.
+		registry
+			.dispatch( CORE_USER )
+			.receiveGetDismissedItems( [
+				RRM_PUBLICATION_APPROVED_OVERLAY_NOTIFICATION,
+			] );
+
+		const { container, waitForRegistry } = render(
+			<ViewContextProvider
+				value={ VIEW_CONTEXT_MAIN_DASHBOARD_VIEW_ONLY }
```

Comment:
We should change this to be the main dashboard as the component would not render in the view-only dashboard even when it is not dismissed.

```suggestion
				value={ VIEW_CONTEXT_MAIN_DASHBOARD }
```